lib/rules/shebang.js

```diff
@@ -13,7 +13,8 @@ const getNpmignore = require("../util/get-npmignore")
 
 const NODE_SHEBANG = "#!/usr/bin/env node\n"
 const SHEBANG_PATTERN = /^(#!.+?)?(\r)?\n/u
-const NODE_SHEBANG_PATTERN = /#!\/usr\/bin\/env node(?: [^\r\n]+?)?\n/u
+const NODE_SHEBANG_PATTERN =
+    /^#!\/usr\/bin\/env(?: -\S+)*(?: \S+=\S+)* node(?: [^\r\n]+?)?\n/u
```

Comment:
`(?: \S+=\S+)* ` has my redos senses tingling.

I think that needs two changes:
1. `(?: [^\s=]+=\S+)* ` we need to remove the `=` from the first matcher otherwise something like `' =' + '=a'.repeat(500) + ' '` will slow things up
2. The replaced `[^\s=]` will also need to exclude a leading `-` in order to not clash with the first group: `(?: -\S+)`, something like `[^\s=-]`

```suggestion
    /^#!\/usr\/bin\/env(?: -\S+)*(?: [^\s=-]+=\S+)* node(?: [^\r\n]+?)?\n/u
```

Reply:
![image](https://github.com/eslint-community/eslint-plugin-n/assets/13050025/d7985e03-e8cd-478c-a6c6-b27b7c8241c7)